console/v1/types_console_plugin.go

```diff
@@ -48,6 +48,26 @@ type ConsolePluginSpec struct {
 	// i18n is the configuration of plugin's localization resources.
 	// +optional
 	I18n ConsolePluginI18n `json:"i18n"`
+	// csp is a list of Content Security Policy directives for the plugin.
+	CSP []ConsolePluginCSP `json:"csp"`
+}
+
+// DirectiveType is an enumeration of supported CSP directives
+type DirectiveType string
+
+const (
+	DefaultSrc DirectiveType = "default-src"
+	ScriptSrc  DirectiveType = "script-src"
+	ImgSrc     DirectiveType = "img-src"
+	StyleSrc   DirectiveType = "style-src"
+	FontSrc    DirectiveType = "font-src"
+	BaseUri    DirectiveType = "base-uri"
```

Comment:
We don't want plugins to be able to change this.

Reply:
@spadgett not sure how they could, since these are enums?

Reply:
Right, we should remove `base-uri` from the enums. It's not something we want to expose to plugins.

Reply:
Oh sorry, I've misunderstood the comment. 
Yes I agree that `base-uri` directive should be only set by console backend.